packages/gcom4/package.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# Copyright 2024 ACCESS-NRI
+# Based on gcom/package.py by scottwales 2023
+# and https://github.com/coecms/access-esm-build-gadi/blob/master/Makefile
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+import llnl.util.filesystem as fs
+
+
+class Gcom4(Package):
+    """
+    GCOM is a wrapper around multiprocessing libraries such as MPI
+    Gcom4 is a package for older versions of GCOM as used by UM vn7.3 in ACCESS ESM 1.5
+    """
+
+    homepage = "https://code.metoffice.gov.uk/trac/gcom"
+    git = "git@github.com:ACCESS-NRI/GCOM.git"
+
+    maintainers = ["penguian"]
+
+    version("4.5", branch="dev")
+    build_directory = join_path("Share", "gcom4.5_access_config")
+
+    variant("mpi", default=True, description="Build with MPI")
+
+    depends_on("fcm", type="build")
+    depends_on("mpi", when="+mpi")
+
+
+    def patch(self):
+        """
+        Perform the equivalent of the following sed commands:
+        sed -i '/build.target{ns}/d' $@/fcm-make/gcom.cfg
+        sed -i 's/-openmp/-qopenmp/g' $@/fcm-make/machines/nci_ifort_openmpi.cfg
+        """
+        with fs.working_dir(self.build_directory):
+            filter_file(
+                r"build\.target\{ns\}.*", "#",
+                join_path("fcm-make", "gcom.cfg"))
+            filter_file(
+                r"-openmp", "-qopenmp",
+                join_path("fcm-make", "machines", "nci_ifort_openmpi.cfg"))
+
+            
+    def install(self, spec, prefix):
```

Comment:
Can you have separate build() function?

Reply:
I will test this.

Reply:
It does not work. Also, [the doc](https://spack.readthedocs.io/en/latest/packaging_guide.html) says 

> `PackageBase` only supports the `install` phase while the `AutotoolsPackage` and `MakefilePackage` support both install and build phases.

I worked around this by calling `self.build()` from `install`. Is that what you want?
```
$ spack install gcom4%intel
[+] /g/data/tm70/pcl851/src/ACCESS-NRI/release/linux-rocky8-x86_64/intel-19.0.5.281/fcm-2021.05.0-i4kamrjix5ggowbmvz26e6a7shh7xxqv
==> openmpi@4.0.2 : has external module in ['openmpi/4.0.2']
[+] /apps/openmpi/4.0.2 (external openmpi-4.0.2-4jwtg3nt7p3zsod4rvn3i3jyyop3arpe)
==> Installing gcom4-4.5-tshdzy43m2k7g6zf3gcysfgrnxcwpkqj
==> No binary for gcom4-4.5-tshdzy43m2k7g6zf3gcysfgrnxcwpkqj found: installing from source
==> Ran patch() for gcom4
==> gcom4: Executing phase: 'install'
==> Warning: Module file /g/data/tm70/pcl851/src/ACCESS-NRI/release/modules/linux-rocky8-x86_64/gcom4/4.5 exists and will not be overwritten
==> gcom4: Successfully installed gcom4-4.5-tshdzy43m2k7g6zf3gcysfgrnxcwpkqj
  Stage: 10.49s.  Install: 1m 37.85s.  Post-install: 0.22s.  Total: 1m 51.09s
  ```

Reply:
I was only wondering if build() was supported. However, what you did makes it clearer.